src/Symfony/Component/DependencyInjection/ContainerBuilder.php

```diff
@@ -486,6 +496,10 @@ public function merge(ContainerBuilder $container)
         $this->addAliases($container->getAliases());
         $this->getParameterBag()->add($container->getParameterBag()->all());
 
+        if (!$this->context->isLocked()) {
+            $this->context->merge($container->getContext());
```

Comment:
But anyway, if we choose to make the `Context` immutable, this lock thing will be removed.